binrw_derive/src/binrw/codegen/write_options/struct_field.rs

```diff
@@ -290,15 +290,10 @@ impl<'a> StructFieldGenerator<'a> {
                     }
                 }
             },
-            FieldMode::Function(_) => {
-                let ty = &self.field.ty;
-                quote! {
-                    let #args = #WRITE_ARGS_TYPE_HINT::<#ty, _, _, _>(
```

Comment:
I thought the purpose of explicitly sending the type through the hint is to get the compiler to complain *here*, instead of somewhere else later, but when I undo this change the diagnostics get significantly worse, so now I am confused about why the field type is explicitly passed. This was introduced in commit 6beeca832cc3381b48cb53d041471d5ee905d7f2. Does anyone remember what the deal is here? Unfortunately, there are *a lot* of missing UI tests to ensure the BinWrite diagnostics remain good.

I don’t see any other issue with this PR. I mean, this apparently isn’t even an issue either, other than that I’d like to figure out what was the deal here. It would be cool to get more BinWrite UI tests but that can be done… later or never. :-)

Reply:
I was also confused about that to be honest. I tried a match statement to use the old code if there was no map/try/repr and the new code if there was, but like you said I got worse diagnostics with the old code doing that. Not sure what's going on here but if you think of something better I'd be happy to change it.